communication/src/main/java/org/jboss/da/communication/pom/PomAnalyzer.java

```diff
@@ -0,0 +1,246 @@
+
+        List<Location> repoLocations = new LinkedList<>();
+        // add local location
+        repoLocations.add(location);
+
+        // add maven central location
+        repoLocations.add(new SimpleLocation("central", "http://repo.maven.apache.org/maven2/"));
+
+        // add all the repositories mentioned in the pom.xml
+        List<SimpleLocation> locationsInPom = getLocationsDefinedInPom(repoDir);
+        repoLocations.addAll(locationsInPom);
+
+        return repoLocations;
+    }
+
+    private List<SimpleLocation> getLocationsDefinedInPom(File repoDir) {
+
+        List<String> pomFilePaths = findAllPomFiles(repoDir);
+        List<SimpleLocation> locations = new LinkedList<>();
+
+        for (String pom : pomFilePaths) {
+            Optional<MavenProject> project = pomReader.analyze(new File(pom));
+            if (project.isPresent() && project.get().getMavenRepositories() != null) {
+                for (MavenRepository repo : project.get().getMavenRepositories()) {
+                    locations.add(new SimpleLocation(repo.getId(), repo.getUrl()));
+                }
+            }
+        }
+
+        return locations;
+    }
+
+    /**
+     * Given the locations of the pom.xml files, setup the tempDir into a structure
+     * where Cartographer will be able to analyze the poms, and return back a
+     * list of ProjectVersionRefs objects generated from the pom.xml files.
+     * @param tempDir
+     * @param poms
+     * @return
+     * @throws TransferException
+     * @throws IOException
+     */
+    private Map<File, ProjectVersionRef> getProjectVersionRefs(File tempDir, List<String> poms)
+            throws TransferException, IOException {
+
+        Map<File, ProjectVersionRef> projectVersionRefs = new HashMap<>();
+
+        for (String pom : poms) {
+            File pomFile = new File(pom);
+            PomPeek peek = new PomPeek(pomFile);
+            projectVersionRefs.put(pomFile.getParentFile().getAbsoluteFile(), peek.getKey());
+
+            String path = ArtifactPathUtils.formatArtifactPath(peek.getKey().asPomArtifact(), carto
+                    .getGalley().getTypeMapper());
+
+            File f = new File(tempDir, path);
+            f.getParentFile().mkdirs();
+            FileUtils.copyFile(pomFile, f);
+        }
+        return projectVersionRefs;
+    }
+
+    /**
+     * Return all the pom.xml paths found under the folderDir directory
+     *
+     * @param folderDir directory to analyze dependencies
+     * @return List of the pom.xml paths
+     */
+    private List<String> findAllPomFiles(File folderDir) {
+
+        Collection<File> pomFilePaths = FileUtils.listFiles(folderDir, new String[]{"xml"}, true);
+
+        List<String> paths = pomFilePaths.stream()
+                                    .filter(f -> f.getName().equals("pom.xml"))
+                                    .map(f -> f.getAbsolutePath())
```

Comment:
Really good point! Will change it! :smiley:  Keep those reviews coming! Really appreciate them!